### ISSUE TITLE
tmerc-cogs Application

### ISSUE THREAD

**Issue by tmercswims** [Twentysix26/Red-Docs#44]
About time I gave this a shot...

List: approved

Repository name: [tmerc-cogs](https://github.com/tmercswims/tmerc-cogs)

Author: tmerc

Description: Various cogs that range from playing sounds and getting cat facts to taking surveys and storing quotes.

**Comment by tmercswims** [comment]
(If you disagree with the approved list, then can this application fall back to one for the beta list?)

**Comment by Kowlin** [comment]
From the small snippets of code that I've read I can somewhat say that the code looks good. However it should be noted that for the "approve" list a very high standard is required

**Comment by Eslyium** [comment]
Alright, Im going to ask you a few questions/make a few comments. Feel free to reply to them.

say - Don't really need

quotes - What makes yours better compared to the others?

randimage - Don't think this would work well unless you make it add linkable pictures.

kz - How does this work exactly?

randomimals - Oh god not another one

voice - We don't need this you can already do this.

catfacts - Iv seen this before on other repos I'm pretty sure.

And as Kowlin said. the "Approved List" Is a high standard. So expect question.

**Comment by tmercswims** [comment]
@Eslyium Thanks for your comments! In response:

**say:** ~~I do not disagree. In fact, I agree.~~ _Edit: Cog has been removed._

**quotes:** I originally created it for two reasons: I wanted the commands to be separate, e.g. `[p]addquote <quote>` is distinct from `[p]quote`, as opposed to [tekulvw's](https://github.com/tekulvw/Squid-Plugins/blob/master/quotes/quotes.py), where they are the same. It also bothered me that quotes did not really have an assigned number, it was just their index in a list, which would change if a quote in the middle was deleted, so mine assigns a persistent number. At the same time, I did not want to impose the restrictions/structure that [calebj's](https://github.com/neonobjclash/calebj-cogs/blob/master/serverquotes/serverquotes.py) has, where it explicitly stores a person to which the quote can be attributed. Unless I am mistaken, I don't think there is a quotes cog that has the same feature set as mine.

**randimage:** I agree with this as well. It was the first cog I wrote, and it has served its singular, original purpose well. I have been meaning to update it to include adding pictures, but have not yet gotten to it. If this feature would be considered necessary for repository approval (even for the beta list), I would prioritize it. _Edit: I have rewritten this cog to make it much more usable. Document on my repository has been updated._

**kz:** KZ is a mod for the game Counter-Strike: Global Offensive. It is essentially parkour, where players try to complete maps as quickly as possible. Additionally, there are a number of different types of skill jumps that players can execute (bunnyhop may be the one that you have heard of). When in-game, players can use commands like `!maptop`, `!jumptop`, etc. in the game chat to see stats about maps, server jump records, etc. This cog aims to mimic that behavior in Discord chat, so one can retrieve these stats outside of the game.

**randimals:** Yeah, I know. If its presence alone is enough to prevent approval, I suppose I can remove it. But if I can make the beta list and keep it, I would prefer that.

**voice:** ~~Yes, you are absolutely right. I will remove this one soon.~~ _Edit: Cog has been removed._

**catfact:** Perhaps. The only one that is part of any of the [currently listed repositories](https://twentysix26.github.io/Red-Docs/red_cog_approved_repos/) is [palmtree5's](https://github.com/palmtree5/palmtree5-cogs/blob/master/catfact/catfact.py), though its behavior is slightly different.

As I said before, if there are issues that would prevent my repository making the approved list, but not the beta list, I would happily be listed as beta.

I also have a question: What is expected of repositories that are listed, whether approved or beta (though I'm sure the expectations are different)? Obviously keep your cogs functional, but is some kind of approval required before adding cogs, or is it just expected that any new cogs' quality will match that of the ones that already exist (which of course I will do)?

**Comment by palmtree5** [comment]
The main difference between catfact here and my catfact is that this one has the ability to get more than 1 at a time. Though if the user doesn't specify a number, the behavior is exactly the same as mine

**Comment by tmercswims** [comment]
This is true.

**Comment by tmercswims** [comment]
So, uh, what's the thinking here? In its updated state, can I make the approved list? If not, is there something else I can do in order to make it? Or if that isn't feasible, is the beta list a possibility?

I guess I'm just unsure if there is something I should do, or if I'm just waiting at this point 😜

**Comment by Redjumpman** [comment]
Let me start by saying that being on the approved list offers no advantages for repo traffic or visibility. Users seem to download whatever they want regardless of the list since they have equal accessibility. In the future maybe something will change, but don't get too caught up in the semantics of the two different lists. The selection process can be a little slow, because staff members and 26 are working on a lot of different issues at once, and they usually try to make a thorough run through of your code for security issues, stealing code without crediting other creators, redundant cogs, correct formatting, and make sure that your stuff works.

Additionally, if you still want to get on the approved list, going from Beta -> Approved is an easier transition after a good deal of time has passed because you can build, improve, and create features that the community wants to increase the popularity of your repo.

TL:DR Be patient these things take time. Approved list is arbitrary. Beta => Approved will be easier for you in the future.

**Comment by tmercswims** [comment]
Thanks for the reply, that helps a lot. I think I was more worried that there was something I should be doing than anything else. I'm not too worried about approved or beta, I just was unsure the status of the thing as a whole, haha.

**Comment by Twentysix26** [comment]
👌 (Beta list for now)

**Comment by tmercswims** [comment]
❤️
